js/libs/ui-shared/src/buttons/FormSubmitButton.tsx

```diff
@@ -0,0 +1,41 @@
+import { Button, ButtonProps } from "@patternfly/react-core";
+import { PropsWithChildren } from "react";
+import { FieldValues, FormState } from "react-hook-form";
+
+export type FormSubmitButtonProps = ButtonProps & {
+  formState: FormState<FieldValues>;
+  allowNonDirty?: boolean;
+  allowInvalid?: boolean;
+};
+
+const isSubmittable = (
+  formState: FormState<FieldValues>,
+  allowNonDirty: boolean,
+  allowInvalid: boolean,
```

Comment:
I feel like if there was code that previously allowed submitting non-dirty and invalid form submissions those should likely be considered bugs. Ideally we don't allow this kind of behavior anywhere, and align this across forms, WDYT?

Reply:
In my opinion, if there is enough visual cue on the required fields or the form is straightforward and obvious, then the button should be disabled on non-dirty and invalid forms.  
However, the required fields (in Keycloak forms) only have a red asterisk attached to the label. In larger forms, they might go unnoticed. In cases like this, an enabled Submit button will be the preferred option as they can click on it and get an immediate feedback of the errors on the form.

That said, I actually prefer enabling the button on non-dirty and invalid form.

Reply:
> However, the required fields (in Keycloak forms) only have a red asterisk attached to the label. In larger forms, they might go unnoticed. In cases like this, an enabled Submit button will be the preferred option as they can click on it and get an immediate feedback of the errors on the form.

That is a good point, ideally we'd be able to attend the user of that by scrolling the invalid input into focus with an error message, but that isn't always possible. This is something that could be improved later on though.

Let's consider this resolved for now and take another look at it when we roll out this behaviour across all the forms.